imap_processing/lo/l1c/lo_l1c.py

```diff
@@ -484,3 +491,98 @@ def create_datasets(
             )
 
     return dataset
+
+
+def set_background_rates(
+    pointing_start_met: float,
+    pointing_end_met: float,
+    species: FilterType,
+    anc_dependencies: list,
+    attr_mgr: ImapCdfAttributes,
+) -> xr.DataArray:
+    """
+    Set the background rates for the specified species.
+
+    The background rates are set to a constant value of 0.01 counts/s for all bins.
+
+    Parameters
+    ----------
+    pointing_start_met : float
+        The start MET time of the pointing.
+    pointing_end_met : float
+        The end MET time of the pointing.
+    species : FilterType
+        The species to set the background rates for. Can be "h" or "o".
+    anc_dependencies : list
+        Ancillary files needed for L1C data product creation.
+    attr_mgr : ImapCdfAttributes
+        Attribute manager used to get the L1C attributes.
+
+    Returns
+    -------
+    background_rates : xarray.DataArray
+        The background rates for the specified species.
+    """
+    if species not in {FilterType.HYDROGEN, FilterType.OXYGEN}:
+        raise ValueError("Species must be 'h' or 'o'.")
+
+    bg_rates = np.zeros((7, 3600, 40), dtype=np.float16)
+    bg_stat_uncert = np.zeros((7, 3600, 40), dtype=np.float16)
+    bg_sys_err = np.zeros((7, 3600, 40), dtype=np.float16)
+
+    # read in the background rates from ancillary file
+    if species == FilterType.HYDROGEN:
+        background_df = lo_ancillary.read_ancillary_file(
+            next(s for s in anc_dependencies if "hydrogen-background" in s)
+        )
+    else:
+        background_df = lo_ancillary.read_ancillary_file(
+            next(s for s in anc_dependencies if "oxygen-background" in s)
+        )
+
+    # find to the rows for the current pointing
+    pointing_bg_df = background_df[
+        (background_df["GoodTime_strt"] >= pointing_start_met)
+        & (background_df["GoodTime_end"] <= pointing_end_met)
+    ]
+
+    # for each row in the bg ancillary file for this pointing
+    for _, row in pointing_bg_df.iterrows():
+        # convert the bin start and end resolution from 6 degrees to .1 degrees
+        bin_start = int(row["bin_strt"]) * 60
+        # The last bin end in the file is 0, which means 60 degrees. This is
+        # converted to 0.1 degree resolution of 3600
+        bin_end = int(row["bin_end"]) * 60
+        bin_end = np.where(bin_end == 0, 3600, bin_end)
```

Comment:
I don't think you need the where here and could just update those indices directly.
```suggestion
        bin_end[bin_end == 0] = 3600
```